src/editor/focus.ts

```diff
@@ -141,7 +141,7 @@ export class Focus {
         this.fireOnNavChangeCallbacks();
     }
 
-    navigatePos(pos: Position) {
+    navigatePos(pos: Position, runNavOffCallbacks: boolean = true) {
```

Comment:
We don't want to run navOffCallbacks twice in here.

These are execution paths that cause this problem:

navigateUp/Down() -> navigatePos()  both run navOffCallbacks because navigatePos() is not always called. On the other hand when we use the mouse navigateUp/Down/Left/Right don't ever get called so it is necessary to have logic that identifies these cases so that we don't run navOff twice.